### PR TITLE
gitbase: implement ref_commits table

### LINKED ISSUE #270: Create new relation table: ref_commits
Ref_commits Table
--
repository_id
commit_hash
ref_name
index

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #270

### CODE REVIEW

File: references.go [hunk 1]
```diff
@@ -25,8 +25,6 @@ func newReferencesTable() sql.Table {
 	return new(referencesTable)
 }
 
-var _ Table = (*referencesTable)(nil)
```
Comment (ajnavarro) [inline references.go]: why did you remove this?

File: ref_commits_test.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+			iter, err := table.WithProjectAndFilters(ctx, nil, tt.filters)
+			require.NoError(err)
+
+			rows, err := sql.RowIterToRows(iter)
+			require.NoError(err)
+
+			for i, row := range rows {
+				// remove blob content and blob size for better diffs
```
Comment (jfontan) [inline ref_commits_test.go]: `// remove repository_id for better diffs`

Comment (jfontan) [reply ref_commits_test.go]: Typo aside LGTM

File: references.go (hunk 1 above)
Comment (erizocosmico) [reply references.go]: It was a mistake, I thought it was `sql.Table` (which is already checked by `var _ sql.PusdownProjectionAndFiltersTable = (*referencesTable)(nil)` but this is checking for the gitbase table interace. I'll re-add that.

**Comment by erizocosmico** [comment]
Fixed

**Comment by erizocosmico** [comment]
@ajnavarro shall we merge this, then?
